CMake checking for std::unique_ptr doesn't honor CXX_STANDARD

found by @robbietuk 

error:
```
 CMake Error at src/CMakeLists.txt:199 (message):
   Problems compiling with std::unique_ptr.  Giving up.
```
Code is
https://github.com/UCL/STIR/blob/e8d250feb8dfbb00fcb25ab63d08282496ac6751/src/CMakeLists.txt#L191-L196

Underlying cause is that the C++ standard hasn't been passed to the compiler when using `check_cxx_source_compiles`.

However, we try to handle this:
https://github.com/UCL/STIR/blob/e8d250feb8dfbb00fcb25ab63d08282496ac6751/src/CMakeLists.txt#L174-L184

This apparently fails for @robbietuk

--- a/src/cmake/SetC++Version.cmake
+++ b/src/cmake/SetC++Version.cmake
@@ -30,6 +30,7 @@ function(UseCXX VERSION)
       endif ()
     else ()
       set (CMAKE_CXX_STANDARD ${VERSION} PARENT_SCOPE)
+      set (CMAKE_CXX_STANDARD_REQUIRED ON PARENT_SCOPE)
     endif ()
 
     message(STATUS "Using C++ version ${VERSION}")

